### PR TITLE
chore: upgrade to go 1.22

### PR DESCRIPTION
**Pull request by lilnasy**
## Changes
- Updates the entrypoint to reflect renaming of the imports object in 1.20.

## Testing
- Existing tests should pass.

## Docs
- Does not affect usage.

### CODE REVIEW

File: packages/compiler/src/node/wasm_exec.ts [hunk 1]
```diff
@@ -165,7 +165,7 @@ export default class Go {
 
     const timeOrigin = Date.now() - performance.now();
     this.importObject = {
-      go: {
+      gojs: {
```
Comment (lilnasy) [inline packages/compiler/src/node/wasm_exec.ts:168]: This is the fix.

**Review by ematipico — APPROVED** [review]

**Review by ematipico — COMMENTED** [review]
Should we update `go.mod` to use `1.22`?

**Comment by lilnasy** [comment]
Yes, we should.

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/withastro/compiler
 
-go 1.19
+go 1.20
```
Comment (ematipico) [inline go.mod:3]:
```suggestion
go 1.22
```

Comment (lilnasy) [reply go.mod:3]: `go.mod` is just the minimum version

Comment (ematipico) [reply go.mod:3]: Why do we require a minimum version, but then build the program in a more recent version? I don't understand

Comment (lilnasy) [reply go.mod:3]: If a contributor has 1.20 installed, it would work the same as 1.22. It's not necessary to make them upgrade their toolchain.

Comment (lilnasy) [reply go.mod:3]: What's the minimum go version needed by the testing library you wanted to use?

Comment (ematipico) [reply go.mod:3]: This left me with more doubts and questions:
- we test and **compile** against a different target. What could work locally might not work on CI and vice versa
- we don't release go libs, why do we require different versions?
- At the moment we all point to 1.19, but now we are branching. Does the contribution guide explain this?

**Review by matthewp — CHANGES_REQUESTED** [review]
If we're going to upgrade the go version we should test against core tests before doing so. It has broken in the past. I would suggest:

1. Creating a preview release here.
2. Creating a PR in core with the compiler version as the preview release version.
3. If tests pass we are golden and can proceed.

**Comment by lilnasy** [comment]
!preview go122

**Comment by lilnasy** [comment]
!preview go122

**Comment by lilnasy** [comment]
All good

**Review by natemoo-re — APPROVED** [review]
Thank you! Preview release passed === LGTM 🫡

File: go.mod (hunk 2 above)
Comment (ematipico) [reply go.mod:3]: Can you answer my questions?

The lib that I need requires 1.21

**Pull request merged**